src/GafferSceneUIModule/RenderPassEditorBinding.cpp

```diff
@@ -0,0 +1,827 @@
+		{
+			setScene( scene );
+			setContext( context );
+		}
+
+		IE_CORE_DECLARERUNTIMETYPEDEXTENSION( RenderPassPath, GafferSceneUI::RenderPassPathTypeId, Gaffer::Path );
+
+		~RenderPassPath() override
+		{
+		}
+
+		void setScene( ScenePlugPtr scene )
+		{
+			if( m_scene == scene )
+			{
+				return;
+			}
+
+			m_scene = scene;
+			m_plugDirtiedConnection = scene->node()->plugDirtiedSignal().connect( boost::bind( &RenderPassPath::plugDirtied, this, ::_1 ) );
+
+			emitPathChanged();
+		}
+
+		ScenePlug *getScene()
+		{
+			return m_scene.get();
+		}
+
+		const ScenePlug *getScene() const
+		{
+			return m_scene.get();
+		}
+
+		void setContext( Gaffer::ContextPtr context )
+		{
+			if( m_context == context )
+			{
+				return;
+			}
+
+			m_context = context;
+			m_contextChangedConnection = context->changedSignal().connect( boost::bind( &RenderPassPath::contextChanged, this, ::_2 ) );
+
+			emitPathChanged();
+		}
+
+		Gaffer::Context *getContext()
+		{
+			return m_context.get();
+		}
+
+		const Gaffer::Context *getContext() const
+		{
+			return m_context.get();
+		}
+
+		bool isValid( const IECore::Canceller *canceller = nullptr ) const override
+		{
+			if( !Path::isValid() )
+			{
+				return false;
+			}
+
+			const PathMatcher p = pathMatcher( canceller );
+			return p.match( names() ) & ( PathMatcher::ExactMatch | PathMatcher::DescendantMatch );
+		}
+
+		bool isLeaf( const IECore::Canceller *canceller ) const override
+		{
+			const PathMatcher p = pathMatcher( canceller );
+			const unsigned match = p.match( names() );
+			return match & PathMatcher::ExactMatch && !( match & PathMatcher::DescendantMatch );
+		}
+
+		PathPtr copy() const override
+		{
+			return new RenderPassPath( m_scene, m_context, names(), root(), const_cast<PathFilter *>( getFilter() ) );
+		}
+
+		void propertyNames( std::vector<IECore::InternedString> &names, const IECore::Canceller *canceller = nullptr ) const override
+		{
+			Path::propertyNames( names, canceller );
+			names.push_back( g_renderPassNamePropertyName );
+			names.push_back( g_renderPassEnabledPropertyName );
+		}
+
+		IECore::ConstRunTimeTypedPtr property( const IECore::InternedString &name, const IECore::Canceller *canceller = nullptr ) const override
+		{
+			if( name == g_renderPassNamePropertyName )
+			{
+				const PathMatcher p = pathMatcher( canceller );
+				if( p.match( names() ) & PathMatcher::ExactMatch )
+				{
+					return new StringData( names().back().string() );
+				}
+			}
+			else if( name == g_renderPassEnabledPropertyName )
+			{
+				const PathMatcher p = pathMatcher( canceller );
+				if( p.match( names() ) & PathMatcher::ExactMatch )
+				{
+					Context::EditableScope scopedContext( getContext() );
+					if( canceller )
+					{
+						scopedContext.setCanceller( canceller );
+					}
+					scopedContext.set( g_renderPassContextName, &( names().back().string() ) );
+					auto enabledData = getScene()->globals()->member<BoolData>( g_renderPassEnabledOption );
```

Comment:
Just spotted this one - another dangling pointer.

Reply:
Thanks! I've tidied that up and squashed it in with the rest of the fixups.